compiler-core/src/ast.rs

```diff
@@ -378,6 +378,8 @@ pub struct Import<PackageName> {
     pub location: SrcSpan,
     pub module: SmolStr,
     pub as_name: Option<SmolStr>,
+    pub as_span: Option<SrcSpan>,
```

Comment:
Let's merge `discarded`, `as_name` and `as_span` into `alias: Option<(AssignName, SrcSpan)>` as it's not valid to have one but not the other.

Reply:
I changed it there and now I'm going all over the code to fix the errors. But I can't figure out how to fix this one(I'm fighting with the borrow checker and just can't win). This is in `src/javascript/typescript.ts` in the `collect_imports` function:
```rust
Definition::Import(Import {
                    module,
                    package,
                    as_name,
                    ..
                }) => {
                    if let Some(alias) = as_name {
                        let _ = self.aliased_module_names.insert(module, alias);
                    }
                    self.register_import(&mut imports, package, module);
                }
``
I've changed it to:
```rust
Definition::Import(Import {
                    module,
                    package,
                    alias,
                    ..
                }) => {
                    if let Some((alias, _)) = alias {
                        let alias: SmolStr = (*alias).clone().into();
                        let _ = self.aliased_module_names.insert(module, alias.as_str());
                    }
                    self.register_import(&mut imports, package, module);
                }
```
(I've implemented `From<AssignName>` for `SmolStr`)
But now I get this error at `alias.as_str()`:
```
'alias' does not live long enough
borrowed value does not live long enough
```
I tried a lot of things but I just can't figure out how to solve this.

Reply:
Could you remove the `From<AssignName>` from `SmolStr` in favour of a pattern matching please. It's not appropriate as `AssignName` contains 2 different names.

The error is that you're cloning the string, so it gets deallocated when the block finishes, leaving a dangling pointer in the hashmap. If you remove the clone it'll work.

Something like this:

```rust
if let Some((AssignName(name), _)) = alias.as_ref() {
    let _ = self.aliased_module_names.insert(module, alias.as_str());
}
```